packages/browser-utils/src/metrics/browserMetrics.ts

```diff
@@ -120,6 +121,45 @@ export function startTrackingLongTasks(): void {
   });
 }
 
+/**
+ * Start tracking long animation frames.
+ */
+export function startTrackingLongAnimationFrames(): void {
+  // NOTE: the current web-vitals version (3.5.2) does not support long-animation-frame, so
+  // we directly observe `long-animation-frame` events instead of through the web-vitals
+  // `observe` helper function.
+  if (PerformanceObserver.supportedEntryTypes.includes('long-animation-frame')) {
+    const observer = new PerformanceObserver(list => {
+      for (const entry of list.getEntries() as PerformanceLongAnimationFrameTiming[]) {
+        if (!getActiveSpan()) {
+          return;
+        }
+        const startTime = msToSec((browserPerformanceTimeOrigin as number) + entry.startTime);
+        const duration = msToSec(entry.duration);
+
+        const span = startInactiveSpan({
+          name: 'Main UI thread blocked',
+          op: 'ui.long-animation-frame',
+          startTime,
+          attributes: {
+            [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.ui.browser.metrics',
+            'script.source_url': entry.scripts[0]?.sourceURL,
+            'script.source_function_name': entry.scripts[0]?.sourceFunctionName,
+            'script.source_char_position': entry.scripts[0]?.sourceCharPosition,
+            'script.invoker': entry.scripts[0]?.invoker,
+            'script.invoker_type': entry.scripts[0]?.invokerType,
```

Comment:
Can we try basing some of these on https://opentelemetry.io/docs/specs/semconv/general/attributes/#source-code-attributes?

```suggestion
            'code.filepath': entry.scripts[0]?.sourceURL,
            'code.function': entry.scripts[0]?.sourceFunctionName,
            'browser.script.source_char_position': entry.scripts[0]?.sourceCharPosition,
            'browser.script.invoker': entry.scripts[0]?.invoker,
            'browser.script.invoker_type': entry.scripts[0]?.invokerType,
```

[`entry.scripts[0]?.sourceCharPosition`](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceScriptTiming/sourceCharPosition) returns `-1` if not found, should we account for this?

We should also avoid using optional chaining, see: https://blog.sentry.io/performance-impact-of-generated-javascript/#removing-optional-chaining

Reply:
Doesn't seem like there is an OpenTelemetry equivalent for invoker/invoker_type. We could get `code.fileno` and `code.column` from the char position but I think it's pretty expensive to perform each time.